staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -1092,8 +1092,8 @@ type SecretVolumeSource struct {
 	// relative and may not contain the '..' path or start with '..'.
 	// +optional
 	Items []KeyToPath `json:"items,omitempty" protobuf:"bytes,2,rep,name=items"`
-	// Optional: mode bits to use on created files by default. Must be a
-	// value between 0 and 0777. Defaults to 0644.
+	// Optional: mode bits used to set permissions on created files by default. 
+	// Must be an octal value between 0 and 0777. Defaults to 0644.
```

Comment:
I think this is a step in the right direction, but should be even clearer to guide people setting values in json/yaml manifests. Can you confirm what happens when you attempt to set these values via the following methods:
1. a yaml manifest file containing an octal number
1. a json manifest file containing an octal number

See related discussion in https://github.com/kubernetes/kubernetes/issues/33162#issuecomment-248651808

If those formats have specific support (or lack of support) for octal numbers, it would be useful to note that here.

Reply:
@liggitt Thanks a lot for taking a look :)
Yaml accepts an octal number but json does not. The json forma does not support octal numbers. See parsing error below:
```
error: rawResources failed to read Resources: json: line 52: invalid character '7' after object key:value pair
```
When I tried converting a yaml to json using an online tool, it auto converted octal numbers to their base 10 equivalent. So, this seemed to further confirm the above. Hope these findings help us move in the right direction.

Reply:
Checking further, schema accepts only int32. (for example: https://github.com/garethr/kubernetes-json-schema/blob/master/master-local/configmapvolumesource-v1.json)
I am not an expert, but if this data type was regex instead of int32, then maybe we could accept octal numbers as well. But this could be more complex, than I can conceive. But, happy to go through the process and learn. Thanks.

Reply:
changing the data type isn't an option. noting that yaml accepts octal or decimal numbers, and json requires decimal numbers would be sufficient.

Reply:
@liggitt,  Sounds good!, updated it. Please take a look.